ui/tooluifactory.cpp

```diff
@@ -50,4 +73,37 @@ bool ToolUiFactory::remotingSupported() const
 
 void ToolUiFactory::initUi()
 {
+
+}
+
+void ToolUiFactory::initUiObject()
+{
+    if (s_ObjectFactoriesInitialized)
+        return;
+    s_ObjectFactoriesInitialized = true;
+
+    PropertyWidget::registerTab<PropertiesTab>(QStringLiteral("properties"),
+                                               ObjectInspectorFactory::tr("Properties"),
+                                               PropertyWidgetTabPriority::First);
+    ObjectBroker::registerClientObjectFactoryCallback<PropertiesExtensionInterface *>(
+        createExtension<PropertiesExtensionClient>);
+    PropertyWidget::registerTab<MethodsTab>(QStringLiteral("methods"),
+                                            ObjectInspectorFactory::tr("Methods"),
+                                            PropertyWidgetTabPriority::Basic - 1);
+    ObjectBroker::registerClientObjectFactoryCallback<MethodsExtensionInterface *>(
+        createExtension<MethodsExtensionClient>);
+    PropertyWidget::registerTab<ConnectionsTab>(QStringLiteral("connections"),
+                                                ObjectInspectorFactory::tr("Connections"),
+                                                PropertyWidgetTabPriority::Basic - 1);
+    ObjectBroker::registerClientObjectFactoryCallback<ConnectionsExtensionInterface *>(
+        createExtension<ConnectionsExtensionClient>);
+    PropertyWidget::registerTab<EnumsTab>(QStringLiteral("enums"),
+                                          ObjectInspectorFactory::tr("Enums"),
+                                          PropertyWidgetTabPriority::Exotic - 1);
+    PropertyWidget::registerTab<ClassInfoTab>(QStringLiteral("classInfo"),
+                                              ObjectInspectorFactory::tr("Class Info"),
+                                              PropertyWidgetTabPriority::Exotic - 1);
+    PropertyWidget::registerTab<ApplicationAttributeTab>(
+                QStringLiteral("applicationAttributes"),
+                ObjectInspectorFactory::tr("Attributes"), PropertyWidgetTabPriority::Advanced);
```

Comment:
The GammaRay code was written with the assumption that the object inspector is always the default / first tool shown.
And this one init most of the tab for any QObject based inspector.
The other inspectors, only init the new one they need (or custom one), because the other were already loaded / initialized by the object inspector.
Now that we can restore the 'initial' tool at startup, the object one is no more the always first shown, so all the tab it does normally init, are not.
So the restored inspectors is missing anything the object inspector is registering.
That's why this change is needed (unless i'm wrong somewhere, which is possible :))